tube/etl/indexers/aggregation/parser.py

```diff
@@ -106,7 +106,39 @@ def get_parent_props(self):
                 )
         return list_nodes
 
+    def add_program_name_to_parent(self):
+        """
+        In case program name is not in self.mapping["parent_props"] while the root node is project. We must add that field
+        :return:
+        """
+        found_program = -1
+        i = -1
+        for path in self.mapping["parent_props"]:
+            p = path.get("path")
+            i += 1
+            if p.startswith("program"):
+                found_program = i
+        if found_program == -1:
+            self.mapping["parent_props"].append(
+                {"path": "programs[{PROGRAM_N}:name]".format(PROGRAM_N=PROGRAM_NAME)}
+            )
+            found_program = len(self.mapping["parent_props"]) - 1
+        program_path = self.mapping["parent_props"][found_program]
+        if program_path["path"].find(PROGRAM_NAME) == -1:
+            program_path += (
+                program_path[: len(program_path) - 1]
+                + "{PROGRAM_N}:name".format(PROGRAM_N=PROGRAM_NAME)
+                + "]"
```

Comment:
Is there a reason to have `+` instead of `format` for everything? Like this:
```
"{PROGRAM_PATH}{PROGRAM_N}:name]".format(PROGRAM_PATH=program_path[: len(program_path) - 1], PROGRAM_N=PROGRAM_NAME)
```